modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -692,6 +685,37 @@ def generate_by_name(by):
                 )
         return new_frame
 
+    def _add_agg_exprs(self, agg, col_to_ref, kwargs, agg_exprs):
+        """
+        Add `AggregateExpr`s for each column to `agg_exprs`.
+
+        Parameters
+        ----------
+        agg : str
+        col_to_ref : dict
+        kwargs : dict
+        agg_exprs : dict
+        """
+        if agg == "nlargest" or agg == "nsmallest":
```

Comment:
I'm afraid this approach won't scale well if we would need more of these functions. Ideally, I see this function as:
```python
def _add_agg_exprs(self, agg, col_to_ref, agg_kwargs, agg_exprs):
    for col, ref in col_to_ref.items():
        agg_exprs[col] = AggregateExpr(agg, ref, agg_kwargs) # and then each aggregation class
                                                             # handles their agg kwargs
```
what do you think?

Reply:
I agree, but AggregateExpr is not a good candidate for this purpose. Currently it's used in many places and with different purposes. A refactoring is definitely required and not only here. But in a separate PR, IMO.

Reply:
ok